collections/nemo_nlp/nemo_nlp/data/datasets/utils.py

```diff
@@ -821,8 +821,13 @@ def __init__(self,
 
 
 class SentenceClassificationDataDesc:
-    def __init__(self, dataset_name, data_dir, do_lower_case):
-        if dataset_name == 'sst-2':
+    def __init__(self, dataset_name, data_dir, do_lower_case, processor=None):
+        # if the user passes a customized data processor
+        if processor is not None:
+            self.data_dir, self.num_labels, self.eval_file, \
+                self.train_file = processor(data_dir)
```

Comment:
do you think users would be passing in train file that isn't self.train_file = self.data_dir + '/train.tsv'?